### ISSUE TITLE
Bugfix: Refine support for LAEA grids via NetCDF and GRIB2 files 

### ISSUE THREAD

**Issue by JohnHalleyGotway** [dtcenter/MET#2565]
## Describe the Problem ##
Development for #1693 ADDED support for reading LAEA grids from GRIB2 files. However, it also REMOVED the existing support for reading LAEA grids from CF-compliant NetCDF files. As such, we commented out the existing unit test that tested that functionality (see `<test name="plot_data_plane_LAEA_NCCF">` in [unit_plot_data_plane.xml](https://github.com/dtcenter/MET/blob/0e29d05c14b842a978880accfaee5e96ea3d340d/internal/test_unit/xml/unit_plot_data_plane.xml#L525)).

This task is to restore that support for reading LAEA data from CF-compliant NetCDF files. Recommend paying careful attention to [Lambert azimuthal equal area](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.10/cf-conventions.html#lambert-azimuthal-equal-area) documentation. And please remember to uncomment the `plot_data_plane_LAEA_NCCF` unit test mentioned above.

Only need to fix this for the `develop` branch prior to the MET-11.1.0 release.

### Expected Behavior ###
Support LAEA grids read from both GRIB2 and CF-complaint NetCDF files.

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Organization** level **Project** for support of the current coordinated release
- [x] Select **Repository** level **Project** for development toward the next official release or add **alert: NEED CYCLE ASSIGNMENT** label
- [x] Select **Milestone** as the next bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
No impacts.

## Bugfix Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Organization** level software support **Project** for the current coordinated release
Select: **Milestone** as the next bugfix version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
Select: **Reviewer(s)** and **Development** issue
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Close this issue.

**Comment by JohnHalleyGotway** [comment]
@mpm-meto and @robdarvell, @rgbullock is working on this issue. He is using a MetOffice data file for testing that is centered on the United Kingdom.
   20230302T2100Z-visibility_at_screen_level.nc

I am wondering if you perhaps have a sample data file that's further away from the prime meridian with which he could also test? For example, in the past you send us data centered over India. Can you provide a sample LAEA CF-Compliant NetCDF file that covers the India domain?

**Comment by mpm-meto** [comment]
I can ask. Perhaps UM partners now running IMPROVER will have some sample data but we don't have any LAEA data over other parts of the world.

**Comment by JohnHalleyGotway** [comment]
OK, thanks for checking. We'll do the best we can with UK LAEA NetCDF sample data we have. If you do get your hands on NetCDF data for additional LAEA domains, please pass it along.

**Comment by JohnHalleyGotway** [comment]
Note that I renamed this issue to clarify that 2 changes are needed:
1. Restore support for parsing LAEA grids from CF-compliant NetCDF files.
2. Refine support for parsing LAEA grids from GRIB2 files.

For the latter, the MET-11.1.0-rc1 version does not properly parse longitudes from the LAEA GRIB2 metadata. The problem lies in the GRIB2C library which MET uses to retrieve that metadata. In the sample GRIB2 files over the United Kingdom, the 2 longitudes defining the LAEA grid have negative values. The GRIB2C library does not properly parse that negative value.

Both 1. and 2. are now fixed. For 2., MET now reprocesses the 4 bytes which constitute the value and reprocess them looking for a negative.
